js/lib/components/feedback/modal.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from "react";
+import Modal from "react-modal";
+import Highlighter from "web-highlighter";
+
+Modal.setAppElement("body");
+
+const modalStyles = {
+  content: {
+    top: "50%",
+    left: "50%",
+    right: "auto",
+    bottom: "auto",
+    marginRight: "-50%",
+    transform: "translate(-50%, -50%)",
+  },
+};
+
+type FeedbackModalProps = { range: Range; highlighter: Highlighter; toggleModal: () => void };
+const FeedbackModal: React.FC<FeedbackModalProps> = ({ range, highlighter, toggleModal }) => {
+  const [feedback, setFeedback] = useState("");
+
+  const handleFeedbackChange = (e: React.FormEvent<HTMLTextAreaElement>) => {
+    setFeedback(e.currentTarget.value);
+  };
+
+  const handleSubmit = () => {
+    // add feedback to serialized highlighter data
+    highlighter.hooks.Serialize.RecordInfo.tap(() => feedback);
```

Comment:
Is there some kind of implicit state variable that this is "tapping" in to? How does this interact with `fromRange`?

Reply:
`tap` is web-highlighter's way of adding middleware for their rendering/serialization methods. Whatever's returned from the callback passed to `RecordInfo.tap` is appended as an `extra` field in the highlight data. When `fromRange` is called, the highlight is serialized and this callback is called, adding `feedback` to the highlight object. I just pushed a change that disposes the "hook" after `fromRange` is called to avoid creating and calling lots of callbacks.